.circleci/tests/test_conn.py

```diff
@@ -0,0 +1,159 @@
+"""Tests for connectivity matrix calculation."""
+
+# Necessary imports
+import fnmatch
+import os
+
+import nibabel as nb
+import nilearn
+import numpy as np
+import pandas as pd
+from nilearn.input_data import NiftiLabelsMasker
+
+from xcp_d.utils.concantenation import _t12native
+from xcp_d.utils.plot import _get_tr
+from xcp_d.utils.write_save import read_ndata, write_ndata
+from xcp_d.workflow.connectivity import init_cifti_conts_wf, init_fcon_ts_wf
+
+
+def nifti_conn_test(data_dir, tmp_path_factory):
+    """Test the nifti workflow."""
+    bold_file = os.path.join(
+        data_dir, "fmriprep/sub-colornest001/ses-1/func/sub-color"
+        "nest001_ses-1_task-rest_run-1_space-MNI152"
+        "NLin2009cAsym_desc-preproc_bold.nii.gz"
+    )
+    bold_mask = os.path.join(
+        data_dir, "fmriprep/sub-colornest001/ses-1/func/"
+        "sub-colornest001_ses-1_task-rest_run-1_space-MNI152"
+        "NLin2009cAsym_desc-brain_mask.nii.gz"
+    )
+    # Generate fake signal
+    bold_data = read_ndata(bold_file, bold_mask)
+    shape = bold_data.shape  # Get the shape so we can generate a matrix of
+    # random numbers with the same shape
+    fake_signal = np.random.randint(bold_data.min(),
+                                    bold_data.max(),
+                                    size=shape)
+    # Let's write that out
+    tempdir = tmp_path_factory.mktemp('fcon_nifti_test')
+    filename = os.path.join(tempdir, "fake_signal_file.nii.gz")
+    write_ndata(
+        fake_signal,
+        template=bold_file,
+        mask=bold_mask,
+        TR=_get_tr(nb.load(bold_file)),
+        filename=filename,
+    )
+    fake_bold_file = filename
+    # Let's define the inputs and create the node
+    mni_to_t1w = os.path.join(
+        data_dir, "fmriprep/sub-colornest001/ses-1/anat"
+        "sub-colornest001_ses-1_rec-refaced_"
+        "from-MNI152NLin2009cAsym_to-T1w_mode-image_xfm.h5"
+    )
+    fcon_ts_wf = init_fcon_ts_wf(
+        mem_gb=4,
+        mni_to_t1w=mni_to_t1w,
+        t1w_to_native=_t12native(bold_file),
+        bold_file=bold_file,
+        name="fcons_ts_wf",
+        omp_nthreads=2,
+    )
+    fcon_ts_wf.inputs.inputnode.clean_bold = fake_bold_file
+    fcon_ts_wf.inputs.inputnode.ref_file = os.path.join(
+        data_dir, "fmriprep/sub-colornest001/ses-1/func"
+        "sub-colornest001_ses-1_task-rest"
+        "_run-1_space-MNI152NLin2009cAsym_boldref.nii.gz"
+    )
+    fcon_ts_wf.base_dir = tmp_path_factory.mktemp("fcon_nifti_test_2")
+    fcon_ts_wf.run()
+    # Let's find the correct FCON matrix file
+    for file in os.listdir(fcon_ts_wf.base_dir + "/fcons_ts_wf/sc47_connect"):
+        if fnmatch.fnmatch(file, "*matrix*"):
+            out_file = file
+    out_file = os.path.join(fcon_ts_wf.base_dir,
+                            "fcons_ts_wf/sc47_connect", out_file)
+    # Read that into a df
+    df = pd.read_csv(out_file, header=None)
```

Comment:
Shouldn't the workflow produce a TSV?

Reply:
at the time it was not - this should be in the "main" pr

Reply:
at the time it was not - this should be in the "main" pr